plugins/catalog/src/translation.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createTranslationRef } from '@backstage/core-plugin-api';
+
+export const catalogTranslationRef = createTranslationRef({
+  id: 'catalog',
+  resources: {
+    en: {
+      catalog: 'catalog-en',
+    },
+    zh: {
+      catalog: '软件目录',
+    },
```

Comment:
I think it's fairly important to have the translations be lazy-loaded for each plugin too. Something like this comes to mind:

```ts
// inside src/translations/zh.ts

export const messages: Translations<typeof catalogTranslationRef> = {
  catalog: '软件目录'
}
```

Then somewhere else we'd have `() => import('./translations/zh.ts')` that would be called if the user selects that language.

The above pattern is something that I'm thinking might make sense to extend to app overrides somehow too.

Reply:
Yes, in plugin level, both support static translations and lazy load translations by use resource or lazyResource in createTranslationRef

Reply:
What benefits do the i18n http-backend bring? Is the goal to have the deployment of message updates be decoupled from frontend deployment?

There may be a risk of overlap with #17054, I want to make sure we don't tangle things too much

Reply:
It's just how i18next works and opt-in, a backend service can be used with a service like weblate or localized so that the translators can work on their own platform instead.

Reply:
Yeah, mostly the message files is located in `packages/app/public/locals` folder

Reply:
Hmm, not sure about the backend in it's current form. It doesn't seem to mesh that well with the build system or TypeScript. Perhaps it's something we can explore as a later addition and sick to lazy-loaded modules for now? Do you have a strong opinion on using the backend plugin as a solution for lazy-loading translations?

Reply:
No strong opinion. It may be that no other concept is introduced except i18n.